### PR TITLE
More flexible structlog - removes opinionated auto configuration

### PR DESCRIPTION
**Pull request by jkebinger**
There are cases where the current opinionated structlog configuration is not appropriate, like any case where a prefab client user needs to configure structlog directly, as in usage with GCP logging, to name one example.

This 
* removes the automatic configuration of structlog, replacing that with an optional method call
* exposes the prefab-specific structlog processors to mix in with other configurations
* documents these in a Readme

### CODE REVIEW

File: prefab_cloud_python/_structlog_processors.py [hunk 1]
```diff
@@ -28,46 +36,50 @@
...
+    event_dict.pop(CallsiteParameter.PATHNAME.value)
+    event_dict.pop(CallsiteParameter.FUNC_NAME.value)
+    event_dict.pop(STRUCTLOG_EVENT_DICT_KEY_PREFAB_CONFIG_CLIENT)
+    event_dict.pop(STRUCTLOG_EVENT_DICT_KEY_LOG_PREFIX)
+    event_dict.pop(STRUCTLOG_EVENT_DICT_KEY_LOG_BOUNDARY)
+    event_dict.pop(STRUCTLOG_EVENT_DICT_KEY_PATH_AGGREGATOR)
+    if STRUCTLOG_EVENT_DICT_KEY_SKIP_AGGREGATOR in event_dict:
+        event_dict.pop(STRUCTLOG_EVENT_DICT_KEY_SKIP_AGGREGATOR)
+    if STRUCTLOG_EVENT_DICT_KEY_INTERNAL_PATH in event_dict:
+        event_dict.pop(STRUCTLOG_EVENT_DICT_KEY_INTERNAL_PATH)
```
Comment (tpetr) [inline prefab_cloud_python/_structlog_processors.py:48]: uber-nit: `event_dict.pop(BLA, None)` won't throw KeyError if BLA is missing, feels a little more succinct

File: prefab_cloud_python/_structlog_processors.py [hunk 2]
```diff
@@ -28,46 +36,50 @@
...
     return event_dict
 
 
 def set_location(_, __, event_dict):
-    if "internal_path" in event_dict:
-        if event_dict["internal_path"]:
-            event_dict["location"] = (
-                "cloud.prefab.client.python.%s" % event_dict["internal_path"]
+    if STRUCTLOG_EVENT_DICT_KEY_INTERNAL_PATH in event_dict:
+        if event_dict[STRUCTLOG_EVENT_DICT_KEY_INTERNAL_PATH]:
```
Comment (tpetr) [inline prefab_cloud_python/_structlog_processors.py:54]: `if event_dict.get(STRUCTLOG_EVENT_DICT_KEY_INTERNAL_PATH):`

File: prefab_cloud_python/constants.py [hunk 3]
```diff
@@ -0,0 +1 @@
+STRUCTLOG_CALLSITE_IGNORES = ["prefab_cloud_python.logger_client"]
```
Comment (tpetr) [inline prefab_cloud_python/constants.py:1]: uber-nit: make this a tuple unless you intend for it to be mutable

File: prefab_cloud_python/structlog_multi_processor.py [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class StructlogMultiProcessor:
```
Comment (tpetr) [inline prefab_cloud_python/structlog_multi_processor.py:1]: can you add a docstring with a tl;dr about what this does / why we need it?

**Review by tpetr — APPROVED** [review]
LGTM, all my python nits are non-binding

**Comment by jkebinger** [comment]
Thank you Tom, i'll take care of these nits shortly

File: prefab_cloud_python/_structlog_processors.py (hunk 2 above)
Comment (jkebinger) [reply prefab_cloud_python/_structlog_processors.py:54]: i realized this is using STRUCTLOG_EVENT_DICT_KEY_INTERNAL_PATH presence, even set to None as an indication that the logging is internal. Added comments. May be cleaner to have an explicit flag but will let it ride for now

**Pull request merged**